packages/core-mobile/app/services/fcm/FCMService.ts

```diff
@@ -0,0 +1,83 @@
+import messaging from '@react-native-firebase/messaging'
+import Logger from 'utils/Logger'
+import NotificationsService from 'services/notifications/NotificationsService'
+import { ChannelId } from 'services/notifications/channels'
+import { ACTIONS, PROTOCOLS } from 'contexts/DeeplinkContext/types'
+import { Dispatch } from '@reduxjs/toolkit'
+import { onFcmTokenChange } from 'store/notifications'
+import { NotificationsBalanceChangeSchema } from 'services/fcm/types'
+
+type UnsubscribeFunc = () => void
+
+/**
+ * Wrapper for @react-native-firebase/messaging
+ */
+class FCMService {
+  getFCMToken = async (): Promise<string> => {
+    return await messaging().getToken()
+  }
+
+  listenForTokenRefresh = (dispatch: Dispatch): void => {
+    messaging().onTokenRefresh(_ => {
+      dispatch(onFcmTokenChange)
+    })
+  }
+
+  listenForMessagesForeground = (): UnsubscribeFunc => {
+    return messaging().onMessage(async remoteMessage => {
+      Logger.info('A new FCM message arrived!', remoteMessage)
+      //TO DESIGN: show in-app notification instead of native
```

Comment:
is this comment still applicable?

Reply:
no, removed in [Skip showing notification if balance changed by sending](https://github.com/ava-labs/core-mobile/pull/1659/commits/9d7620655c30af255e276d1fdf9f3c53d5c6c7a9)